source/common/stats/thread_local_store.cc

```diff
@@ -471,9 +488,11 @@ Histogram& ThreadLocalStoreImpl::ScopeImpl::histogramFromStatName(StatName name,
   // a temporary, and address sanitization errors would follow. Instead we must
   // do a find() first, using that if it succeeds. If it fails, then after we
   // construct the stat we can insert it into the required maps.
-  Stats::SymbolTable::StoragePtr final_name = symbolTable().join({prefix_.statName(), name});
-  StatName final_stat_name(final_name.get());
 
+  auto stat_name_no_tags = StatName(symbolTable().join({prefix_.statName(), name}).get());
```

Comment:
store the return StoragePtr in a temp that outlives stat_name_no_tags.

Reply:
I'd recommend also not using 'auto' here.

Reply:
one more thing here, just for your intuition: `StatName` is a lot like `string_view`; you would probably not want to do:

 absl::string_view view(std::string("xx"));

because the std::string would be destructed before you could use the view.

Reply:
Got it, I think my confusion comes from the fact that it seems like it's okay to destruct the `Storage` after the stat name is handed to the store/scope, but I'm guessing the store takes ownership over that in some sense, or perhaps copies it?

Reply:
copies it.